### PR TITLE
[Playground] Healthcheck was added

### PR DESCRIPTION
**Pull request by MakarkinSAkvelon**
solves #24310
Add Healthcheck and Autoscaling in Kubernetes deployments

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by MakarkinSAkvelon** [comment]
R: @olehborysevych

**Comment by MakarkinSAkvelon** [comment]
R: @rshamunov

**Review by rshamunov — APPROVED** [review]
LGTM

**Comment by olehborysevych** [comment]
@rshamunov @MakarkinSAkvelon I don't think this is correct. At least for backend services. We have two endpoints in our backend services - /liveness and /readiness

**Comment by olehborysevych** [comment]
LGTM

**Comment by MakarkinSAkvelon** [comment]
R: @damondouglas

File: playground/infrastructure/helm-playground/templates/autoscaling-java.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: playground-java
+spec:
+  maxReplicas: 10
+  minReplicas: 2
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/autoscaling-java.yaml:21]: Why is this one configured with `2` while the others `1`?

File: playground/infrastructure/helm-playground/templates/deployment-frontend.yml [hunk 2]
```diff
@@ -34,6 +34,25 @@ spec:
...
+       livenessProbe:
+         httpGet:
+           path: /liveness
+           port: 8080
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       readinessProbe:
+         httpGet:
+           path: /readiness
+           port: 8080
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-frontend.yml:46]: Is it possible for the `8080` to be acquired from a helm variable to populate `ports.containerPort`, `livenessProbe.httpGet.port`, and `readinessProbe.httpGet.port`?

File: playground/infrastructure/helm-playground/templates/deployment-frontend.yml [hunk 3]
```diff
@@ -34,6 +34,25 @@ spec:
...
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       resources:
+         requests:
+           memory: "50Mi"
+           cpu: "250m"
+         limits:
+           cpu: "500"
+           memory: "500Mi"
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-frontend.yml:55]: How was the values `50Mi`, `250m`, `500`, and `500Mi` determined?  I wouldn't know myself and not a blocker for this PR.  I was just curious and thank you.

File: playground/infrastructure/helm-playground/templates/deployment-go.yml [hunk 4]
```diff
@@ -39,6 +39,27 @@ spec:
...
+       livenessProbe:
+         httpGet:
+           path: /liveness
+           port: 8080
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       readinessProbe:
+         httpGet:
+           path: /readiness
+           port: 8080
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-go.yml:53]: Similar to the comment above, can we populate the port parameter from a helm variable?

File: playground/infrastructure/helm-playground/templates/deployment-java.yml [hunk 5]
```diff
@@ -38,6 +38,27 @@ spec:
...
+       livenessProbe:
+         httpGet:
+           path: /liveness
+           port: 8080
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       readinessProbe:
+         httpGet:
+           path: /readiness
+           port: 8080
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-java.yml:52]: Similar to the comment above, can we populate the port parameter from a helm variable?

File: playground/infrastructure/helm-playground/templates/deployment-python.yml [hunk 6]
```diff
@@ -38,6 +38,27 @@ spec:
...
+       livenessProbe:
+         httpGet:
+           path: /liveness
+           port: 8080
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       readinessProbe:
+         httpGet:
+           path: /readiness
+           port: 8080
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-python.yml:52]: Similar to the comment above, can we populate the port parameter from a helm variable?

File: playground/infrastructure/helm-playground/templates/deployment-router.yml [hunk 7]
```diff
@@ -40,6 +40,27 @@ spec:
...
+       livenessProbe:
+         httpGet:
+           path: /liveness
+           port: 8080
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       readinessProbe:
+         httpGet:
+           path: /readiness
+           port: 8080
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-router.yml:54]: Similar to the comment above, can we populate the port parameter from a helm variable?

File: playground/infrastructure/helm-playground/templates/deployment-scio.yml [hunk 8]
```diff
@@ -37,6 +37,27 @@ spec:
...
+       livenessProbe:
+         httpGet:
+           path: /liveness
+           port: 8080
+         initialDelaySeconds: 3
+         periodSeconds: 3
+       readinessProbe:
+         httpGet:
+           path: /readiness
+           port: 8080
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/templates/deployment-scio.yml:51]: Similar to the comment above, can we populate the port parameter from a helm variable?

File: playground/infrastructure/helm-playground/values.yaml [hunk 9]
```diff
@@ -29,4 +29,12 @@ autoscaling:
...
-   targetCPUUtilizationPercentage: 80
\ No newline at end of file
+   targetCPUUtilizationPercentage: 80
+static_ip: 34.120.95.76
+redis_ip: 10.226.29.100:6379
+project_id: astst-369706
+registry: us-east1-docker.pkg.dev/astst-369706/playground-repository
+static_ip_name: pg-static-ip
+tag: teta
+dns_name: dev-playground.online
```
Comment (damondouglas) [inline playground/infrastructure/helm-playground/values.yaml:39]: Do we want to hardcode these?

**Review by damondouglas — CHANGES_REQUESTED** [review]
@MakarkinSAkvelon

File: playground/infrastructure/helm-playground/templates/autoscaling-java.yaml (hunk 1 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/autoscaling-java.yaml:21]: We tested Java backend. For normal autoscaling at least +1 pod required

File: playground/infrastructure/helm-playground/templates/deployment-frontend.yml (hunk 2 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-frontend.yml]: Fixed. Everything was moved to the values.yaml file

File: playground/infrastructure/helm-playground/templates/deployment-go.yml (hunk 4 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-go.yml]: Fixed. Everything was moved to the values.yaml file

File: playground/infrastructure/helm-playground/templates/deployment-java.yml (hunk 5 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-java.yml]: Fixed. Everything was moved to the values.yaml file

File: playground/infrastructure/helm-playground/templates/deployment-python.yml (hunk 6 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-python.yml]: Fixed. Everything was moved to the values.yaml file

File: playground/infrastructure/helm-playground/templates/deployment-router.yml (hunk 7 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-router.yml]: Fixed. Everything was moved to the values.yaml file

File: playground/infrastructure/helm-playground/templates/deployment-scio.yml (hunk 8 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-scio.yml]: Fixed. Everything was moved to the values.yaml file

File: playground/infrastructure/helm-playground/values.yaml (hunk 9 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/values.yaml]: Those variables are changing automatically thru the Gradle task (if they exist) or will be created during execution. Removed for now

File: playground/infrastructure/helm-playground/templates/deployment-frontend.yml (hunk 3 above)
Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/deployment-frontend.yml:55]: We did some tests thru the Apache JMeter. These resources are enough for simple tasks, and they will be autoscaled once loading will be increased

**Review by damondouglas — APPROVED** [review]
@pabloem LGTM

**Comment by pabloem** [comment]
lgtm thanks!

**Pull request merged**